### DISCUSSION TITLE
Documentation Request: Differences from DATA-DOG/go-sqlmock?

### DISCUSSION THREAD

**Discussion by StevenACoffman** [pashagolub/pgxmock#82] [General]
Hello!
We are currently using [DATA-DOG/go-sqlmock](https://github.com/DATA-DOG/go-sqlmock) for Go applications that connect to PostgreSQL database via pgx. I noticed this library was forked from go-sqlmock, but I'm not quite sure _when_ it was forked, nor what was changed in either library happened in the meantime. I noticed pgxmock is missing [column.go](https://github.com/DATA-DOG/go-sqlmock/blob/master/column.go) so maybe [Column Metadata](https://github.com/DATA-DOG/go-sqlmock/pull/152) was merged in sqlmock after this forked? However, [statement.go](https://github.com/DATA-DOG/go-sqlmock/blob/master/statement.go) is also missing, but probably that file was deliberately deleted?

I _think_ pgxmock implements some new pgx-specific methods like:
```
	ExpectBeginTx(txOptions pgx.TxOptions) *ExpectedBegin
	ExpectCopyFrom(expectedTableName string, expectedColumns []string) *ExpectedCopyFrom
	NewRowsWithColumnDefinition(columns ...pgproto3.FieldDescription) *Rows
	NewColumn(name string) *pgproto3.FieldDescription
```
Are there any other features added to pgxmock that I should be aware of?

**Comment by pashagolub** [comment]
Hey!

Thanks a lot for your interest in this project!

The complete change log would be [here](https://github.com/pashagolub/pgxmock/compare/v1.0.0...v1.6.0). 

You shouldn't be afraid of changes to `pgxmock` compared to `sqlmock`. I did everything to be backward compatible with the `sqlmock`. Usually, changes are just enhancements because `pgx` is a far more advanced library than the standard one.

Yes, some files were merged into one. For example, I'm not supporting ancient Go versions, thus I can merge some functionality into one file. 

Anyway, I would appreciate any feedback. Just let me know what areas need improvements, and we will fix that.

Kind regards
